File: PHP/valid_braces.php
<?php

function validBraces($braces){
    $stack = [];
    $match = array(')'=>'(','}'=>'{',']'=>'[');
    for($i=0;$i<strlen($braces);$i++){
        $char = $braces[$i];
        if(in_array($char,['(','{','['])){
            array_push($stack,$char);
        }elseif(in_array($char,[')',']','}'])){
            if(empty($stack) || $stack[count($stack)-1] != $match[$char]){
                return false;
            }else{
                array_pop($stack);
            }
        }
    return empty($stack);
    }

}

File: Python/valid_braces.py
def valid_braces(string):
    stack = []
    brace_match = {')':'(','}':'{',']':'['}
    for char in string :
        if char in '({[':
            stack.append(char)
        elif char in ')}]':
            if not stack or stack[-1] != brace_match[char]:
                return False 
            else:
                stack.pop()
    return not stack
